frontend/components/feedback-task/SearchBar.base.vue

```diff
@@ -16,87 +16,99 @@
   -->
 
 <template>
-  <form @submit.prevent="searchText(query)">
-    <div class="searchbar" :class="{ active: query }" :style="searchBarStyles">
-      <BaseIcon
-        v-if="!query"
-        icon-name="search"
-        icon-width="20"
-        icon-height="20"
-      />
-      <BaseButton
-        v-else
-        class="searchbar__button"
-        @click="removeCurrentSearchText()"
-      >
-        <BaseIcon
-          class="searchbar__button__icon"
-          icon-name="close"
-          icon-width="20"
-          icon-height="20"
-        />
-      </BaseButton>
-      <label class="searchbar__label" for="query" v-text="description" />
-      <input
-        ref="input"
-        class="searchbar__input"
-        type="text"
-        name="query"
-        id="query"
-        v-model.lazy="query"
-        :placeholder="placeholder"
-        autocomplete="off"
-      />
-    </div>
-  </form>
+  <div
+    class="search-area"
+    :class="{ active: isSearchActive }"
+    @click="focusInSearch"
+  >
+    <BaseIconWithBadge
+      class="searchbar__icon"
+      :icon="iconType"
+      :show-badge="false"
+      iconColor="#acacac"
+      badge-vertical-position="top"
+      badge-horizontal-position="right"
+      badge-border-color="white"
+      @click-icon="resetValue"
+    />
+    <input
+      ref="searchRef"
+      class="searchbar__input"
+      type="text"
+      v-model.trim="searchValue"
+      :placeholder="placeholder"
+      :aria-description="description"
+      autocomplete="off"
+      @keydown.enter.exact="applySearch"
+      @keydown.arrow-right.stop=""
+      @keydown.arrow-left.stop=""
+      @keydown.delete.exact.stop=""
+      @keydown.enter.exact.stop=""
+    />
+  </div>
 </template>
 
 <script>
 export default {
+  name: "SearchBarComponent",
   props: {
-    currentSearchText: {
+    value: {
       type: String,
-      default: "",
+      default: () => "",
```

Comment:
The prop works without arrow function for String, there is a purpose?

Reply:
We should be consistent with the props definition. We should apply the same way of default props definition (I mean if we decide to use arrow functions, we should define all props defaults as arrow functions for primitive types, if we choose to use the value directly, we should apply this to all default prop definitions.